docker/Dockerfile.pdo-tp-ccf

```diff
@@ -60,7 +61,8 @@ ENV PDO_INSTALL_ROOT=/pdo/dev/
 ENV PDO_SOURCE_ROOT=/pdo/private-data-objects/
 ENV PDO_ENCLAVE_CODE_SIGN_PEM=/pdo/private-data-objects/ccf_transaction_processor/enclave.pem
 ENV PDO_LEDGER_KEY_ROOT=/pdo/dev/opt/pdo/ccf/keys/
-ENV SGX_MODE=SIM
+ARG SGX_MODE=SIM
+ENV SGX_MODE=${SGX_MODE}
```

Comment:
I believe this is _not_ for running the pdo-tp in HW mode, but rather just for setting the `CHECK_ATTESTATION` compile flag.
If so, would you add a comment?
Another question: in Sawtooth, we have a nice option to run the tp with the `debug-on` option when we are in SIM mode, so that it accepts an empty proof data. Is there a similar run-time mechanism in CCF (I think it's preferable, as opposed to a compile-time one)?

Reply:
thank you, correct, currently there is no support to run pdo tp itself in HW mode. As you say, the SGX_MODE flag is to determine where attestation needs to be checked or not. Decided not to create a new "CHECK ATTESTATION" env, rather use SGX_MODE itself, since that made sense. 

Yes, I see that the Sawtooth ignores attestation with debug-on, I think what we do here with makes more sense to me (to check attestation whenever SGX_MODE =HW). I don't see why we need a run-time flag just for attestation check within the tp. such an argument then can pretty much be extended to every check within the tp.

Reply:
So this should be fixed using the ~/build/cmake/SGX.cmake which will set the HW variables we need. If there are variables missing in there, then add them. We're adding a bunch of additional random places to change configurations.